web/ts/TUMLiveVjs.ts

```diff
@@ -121,6 +123,81 @@ const skipSilence = function (options) {
     });
 };
 
+
+/**
+ * @function watchProgress
+ * Saves and retrieves the watch progress of the user as a fraction of the total watch time
+ * @param streamID The ID of the currently watched stream
+ */
+const watchProgress = function (streamID: number) {
+    this.ready(() => {
+        postData("/api/progressRequest", {
+            "streamID": streamID,
+        }).then((data) => {
+            if (data.status !== 200) {
+                console.log(data);
+            } else {
+                data.text().then(data => {
+                    const json = JSON.parse(data);
+                    this.progressRatio = json["progress"];
+                })
+            }
+        });
+
+        let initialized = false;
+
+        // Fetch the user's stream progress from the database and set the time on load
+        this.on('loadedmetadata', () => {
+            if (initialized) {
+                return;
+            }
+            this.currentTime(this.progressRatio * this.duration());
+            initialized = true;
+        });
+
+        // iPhone/iPad need to play the video first, so they depend on a different event
+        // More info: https://www.w3.org/TR/html5/embedded-content-0.html#mediaevents
+        this.on("canplaythrough", () => {
+            if (initialized) {
+                return;
+            }
+            this.currentTime(this.progressRatio * this.duration());
+            initialized = true;
+        });
+
+        let lastChecked = new Date();
+
+        // Fetch the user's stream progress from the database and set the time on load
+        this.on('timeupdate', () => {
+            const now = new Date();
+
+            const diff = now - lastChecked;
+
+            // Proceed with progress report every 10 seconds
+            if (diff.valueOf() / 1000 < 10) {
+                return;
+            }
+
+            const ctime = this.currentTime();
+            const duration = this.duration();
+            const progress = ctime / duration;
+
+            postData("/api/progressReport", {
+                "streamID": streamID,
+                "progress": progress
+            }).then(r => {
+                    if (r.status !== 200) {
+                        console.log(r);
+                    }
```

Comment:
It would be nice to exponentially back off when the request fails, this way I can control possible load issues using traefik rate limits 
e.g.:
10 seconds -> request -> fail -> 20 seconds -> fail -> 40 seconds ...

Reply:
Agreed, this should drastically increase robustness.